src/Controllers/Events/editEvent.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+use MyRadio\MyRadio\AuthUtils;
+use MyRadio\MyRadio\URLUtils;
+use MyRadio\ServiceAPI\MyRadio_Event;
+use MyRadio\ServiceAPI\MyRadio_User;
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+    $data = MyRadio_Event::getForm()->readValues();
+    if (empty($data['id'])) {
+        // create new
+        $event = MyRadio_Event::create($data);
+        URLUtils::redirectWithMessage(
+            'Events',
+            'viewEvent',
+            'Your event has been created.',
+            [
+                'eventid' => $event->getID()
+            ]
+        );
+    } else {
+        // edit
+        /** @var MyRadio_Event $event */
+        $event = MyRadio_Event::getInstance($data['id']);
+
+        // check permissions
+        $event->checkEditPermissions();
+
+        if ($data['title'] !== $event->getTitle()) {
+            $event->setTitle($data['title']);
```

Comment:
Couldn't we just have an "update all", even if it passes some same-as-before info, to take this down to just 1 db hit

Reply:
Good point, will fix